chain/network/src/peer_manager.rs

```diff
@@ -333,15 +383,32 @@ impl PeerManagerActor {
             .spawn(ctx);
     }
 
+    fn announce_account(&mut self, ctx: &mut Context<Self>, announce_account: AnnounceAccount) {
+        // If this is a new account send an announcement to random set of peers.
+        if self.routing_table.update(&announce_account).is_new() {
+            let msg = SendMessage { message: PeerMessage::AnnounceAccount(announce_account) };
+
+            // TODO(MarX): Don't send to all peers. Sample a subset of K peers
+            // TODO(MarX): Add k to configuration.
+            let requests: Vec<_> =
```

Comment:
there `self.broadcast_message` that does this this.
 you can also later add `self.broadcast_message_random_peers()`